src/common/pool_map.c

```diff
@@ -1551,8 +1553,19 @@ add_domain_tree_to_pool_buf(struct pool_map *map, struct pool_buf *map_buf,
 
 	ordered_ranks->rl_nr = i;
 
-	if (rc == 0 && !updated)
-		return -DER_ALREADY;
+	if (rc == 0) {
+		/* Those ranks already exists in the pool map, but some of them are new, so
+		 * still need extend the pool, though does not need update the pool map
+		 */
+		if (!updated && found_new_dom)
+			return -DER_EXIST;
+
+		/* Those ranks already exists in the pool map, and they are not new, then return
+		 * EALREADY
+		 */
+		if (!updated && !found_new_dom)
+			return -DER_ALREADY;
```

Comment:
[Question] Does returning -DER_ALREADY mean that we don't need to update the pool map? I can't find anywhere handling this -DER_ALREADY.

Reply:
oh, yes, it means the target can not be added. DER_ALREADY will actually return to the user to notify the user all target has been added to the pool map, so "dmg pool extend" failed.